src/Libraries/Microsoft.AspNetCore.Telemetry.Middleware/Logging/HttpLoggingServiceExtensions.cs

```diff
@@ -1,77 +1,123 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#if NET8_0_OR_GREATER
+
 using System;
+using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
-using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.HttpLogging;
 using Microsoft.AspNetCore.Telemetry.Http.Logging;
 using Microsoft.AspNetCore.Telemetry.Internal;
+using Microsoft.Extensions.Compliance.Classification;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
-using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Http.Telemetry;
 using Microsoft.Extensions.Telemetry.Internal;
-using Microsoft.IO;
+using Microsoft.Shared.DiagnosticIds;
 using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.AspNetCore.Telemetry;
 
 /// <summary>
 /// Extension methods to register the HTTP logging feature within the service.
 /// </summary>
+[Experimental(diagnosticId: Experiments.HttpLogging, UrlFormat = Experiments.UrlFormat)]
 public static class HttpLoggingServiceExtensions
 {
     /// <summary>
-    /// Adds components for incoming HTTP requests logging into <see cref="IServiceCollection"/>.
+    /// Enables redaction of HTTP logging.
     /// </summary>
-    /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
-    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    /// <param name="services">The service collection.</param>
+    /// <param name="configureRedaction">Configures the redaction options.</param>
+    /// <param name="configureLogging">Configures the logging options.</param>
+    /// <returns>The original service collection.</returns>
     /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null" />.</exception>
-    public static IServiceCollection AddHttpLogging(this IServiceCollection services)
+    public static IServiceCollection AddHttpLoggingRedaction(this IServiceCollection services, Action<LoggingRedactionOptions>? configureRedaction = null,
+        Action<HttpLoggingOptions>? configureLogging = null)
```

Comment:
Having two callbacks here is awkward, but slightly more discoverable than not having them.

Ideas:
- Don't combine, doc that AddHttpLogging controls half of the settings.
- Duplicate the options from HttpLoggingOptions and then copy them over? That's especially awkward with the header collections that come pre-populated. (Need to consolidate the those settings too).